### PR TITLE
 feat: [PIE-2792]: azure connector setup

### PR DESCRIPTION
**Pull request by Tamaraius**
##### Summary:

Azure Connector setup 

##### Jira Links:

https://harness.atlassian.net/browse/PIE-2792 credential flow
https://harness.atlassian.net/browse/PIE-3214 managed identity flow
cd-ng azure service setup 

##### Screenshots:
https://user-images.githubusercontent.com/96610348/161227879-413da519-d591-46d9-b841-1a940a4150da.mov

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Review by satyam-harness — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+
+    await act(async () => {
+      fireEvent.change(container.querySelector('input[name="name"]')!, {
+        target: { value: 'dummy name' }
+      })
+    })
+    expect(container).toMatchSnapshot()
+
+    await act(async () => {
+      fireEvent.click(container.querySelector('button[type="submit"]')!)
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:198]: Let's changes these querySelector to getByText.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+    })
+    expect(container).toMatchSnapshot()
+
+    await act(async () => {
+      fireEvent.click(container.querySelector('button[type="submit"]')!)
```
Comment (chetanvekariya-harness) [inline src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:203]: Here as well.

Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:203]: will be changed later, thanks!

File: src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx (hunk 1 above)
Comment (Tamaraius) [reply src/modules/35-connectors/components/CreateConnector/AzureConnector/__tests__/CreateAzureConnector.test.tsx:198]: will be changed latter, thanks!

**Review by kapilkumawat86 — APPROVED** [review]

**Pull request merged**